server/src/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -325,7 +325,7 @@ public TemplateProfile prepare(RegisterIsoCmd cmd) throws ResourceAllocationExce
         Long zoneId = cmd.getZoneId();
         // ignore passed zoneId if we are using region wide image store
         List<ImageStoreVO> stores = _imgStoreDao.findRegionImageStores();
-        if (!(stores != null && stores.size() > 0)) {
+        if (!(stores != null && stores.size() > 0) && zoneId != null && zoneId != -1L) {
```

Comment:
You can do zoneId>0 instead of != - 1

Reply:
`CollectionsUtils` here is also a good idea.

Reply:
Done, sorry I didn't get the use of CollectionUtils on this, can you elaborate @rafaelweingartner?